recipe/migrations/cuda120.yaml

```diff
@@ -0,0 +1,132 @@
+migrator_ts: 1682985063
+__migrator:
+  kind:
+    version
+  migration_number:
+    1
+  build_number:
+    1
+  paused: false
+  override_cbc_keys:
+    - cuda_compiler_stub
+  operation: key_add
+  ordering:
+    cxx_compiler_version:
+      - 9
+      - 8
+      - 7
+    c_compiler_version:
+      - 9
+      - 8
+      - 7
+    docker_image:
+      - quay.io/condaforge/linux-anvil-comp7        # [os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-aarch64      # [os.environ.get("BUILD_PLATFORM") == "linux-aarch64"]
+      - quay.io/condaforge/linux-anvil-ppc64le      # [os.environ.get("BUILD_PLATFORM") == "linux-ppc64le"]
+      - quay.io/condaforge/linux-anvil-armv7l       # [os.environ.get("BUILD_PLATFORM") == "linux-armv7l"]
+      - quay.io/condaforge/linux-anvil-cuda:9.2     # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-cuda:11.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+      - quay.io/condaforge/linux-anvil-comp7        # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+    cuda_compiler_version:
+      - None
+      - 10.2                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.0                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.1                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.2                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 12.0                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  commit_message: "Rebuild for CUDA 12"
+
+cuda_compiler:                 # [linux64 or win]
+  - nvcc                       # [linux64 or win]
+  - nvcc                       # [linux64 or win]
+  - nvcc                       # [linux64 or win]
+  - nvcc                       # [linux64 or win]
```

Comment:
Think we will want to add [these values in `conda_build_config.yaml`]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/c67a32ec7d972c1d92f2f26e3fefc6f43f644c56/recipe/conda_build_config.yaml#L44-L45 ) as well

Reply:
But that would be after the migration, right?

Reply:
We can make some of these changes to `conda_build_config.yaml` even before the migrator. This specific change shouldn't affect the rendering of existing feedstocks (except maybe to update [the `zip_keys` entry in variants]( https://github.com/conda-forge/ucx-split-feedstock/blob/8a97995a5cd61a740bbe5a82e55b946fc8399621/.ci_support/linux_64_c_compiler_version9cdt_namecos7cuda_compiler_version11.0cxx_compiler_version9.yaml#L25-L30 ), which shouldn't otherwise affect behavior)